### PR TITLE
Add editor config file

### PR DESCRIPTION
**Pull request by dawehner**

### CODE REVIEW

**Comment by martinfrances107** [comment]
Just to bind two issues together

As typescript is becoming a thing.  I think we are about to make use of autofixing 
I have been thinking about this as part of Linting: Check ts and tsx file ( https://github.com/jsdrupal/drupal-admin-ui/issues/539 ) 

Where the existing rules coded into .eslintrc - are being worked into a yet to be created  tslint.json and tsconfig.json. Where the rules I was thinking about extending  a standard ruleset for example "tslint:recommeded" and "tslint-config-prettier"

I am happy to see this PR be merged 

I just want to say in public is that we/I may have some reconciliation work, after the fact,   so that the coding standard rules seen by the IDE ( those affected by this config file ) and those imposed by the combination of tsconfig.json and tslint.json all work together.

In short this is ready to be merged.
